### PR TITLE
Introduce a SplitRegistry model to avoid passing around hashes

### PR DESCRIPTION
**Pull request by aburgel**
/domain @Betterment/test_track_core 
/no-platform

A follow on to #82 (and mostly based on @jmileham's work that he reverted in #58). ~~I'm keeping this in draft mode until #82 lands. I think I need to add a few more specs too.~~

TODO
- [x] Run against an app's test suite

### CODE REVIEW

File: app/models/test_track/split_registry.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  def split_names
+    registry_hash['splits'].keys
+  end
+
+  def experience_sampling_weight
+    registry_hash.fetch('experience_sampling_weight')
+  end
+
+  def weights_for(split_name)
+    registry_hash && registry_hash['splits'][split_name] && registry_hash['splits'][split_name]['weights']
```
Comment (aburgel) [inline app/models/test_track/split_registry.rb]: TODO: freeze the weights hash to be a little safer

Comment (jmileham) [reply app/models/test_track/split_registry.rb]: like we could freeze the entire registry_hash on ingest, I figure, right?

Comment (jmileham) [reply app/models/test_track/split_registry.rb]: (deep freeze)

**Comment by aburgel** [comment]
@Betterment/test_track_core this is ready for review

**Comment by aburgel** [comment]
@jmileham ran this against a few apps and their test suites, all looks good. that mystery guest issue seems to have been resolved.

**Comment by jmileham** [comment]
<< domain lgtm! Thank you!

**Pull request merged**